src/mirror.rs

```diff
@@ -0,0 +1,65 @@
+use url::{Url, ParseError};
+use std::collections::HashSet;
+
+pub struct MirrorContext {
+    base_urls: Vec<Url>,
+}
+
+impl MirrorContext {
```

Comment:
How about calling that `BaseUrl` or `Base` or `Context` and to use it to resolve relative pathes with?

That you can provide a list of Mirrors instead of just one base url can pretty well be a implementation detail in most cases.